sktime/transformations/series/detrend/_detrend.py

```diff
@@ -107,10 +117,22 @@ def transform(self, Z, X=None):
             De-trended series
         """
         self.check_is_fitted()
-        z = check_series(Z, enforce_univariate=True)
+        z = check_series(Z)
         fh = ForecastingHorizon(z.index, is_relative=False)
-        z_pred = self.forecaster_.predict(fh, X)
-        return z - z_pred
+
+        # multivariate
+        if isinstance(z, pd.DataFrame):
+            # check if all columns are known
+            if not all(k in self.forecaster_ for k in z.columns):
```

Comment:
for the check or printing missing columns, you could do sth like `set(z.columns).difference(set(self.forecaster_.keys()))`; for readability you could first define variables `Z_fit_keys = set(self.forecaster_.keys())`, `Z_new_keys = ...`

Reply:
Thanks! I implemented it as you suggested